internal/nfs/controller/volume.go

```diff
@@ -169,22 +202,49 @@ func (nv *NFSVolume) DeleteExport() error {
 		return fmt.Errorf("failed to identify NFS cluster: %w", err)
 	}
 
-	// ceph nfs export rm <cluster_id> <pseudo_path>
-	args := []string{
-		"--id", nv.cr.ID,
-		"--keyfile=" + nv.cr.KeyFile,
-		"-m", nv.mons,
-		"nfs",
-		"export",
-		"delete",
-		nfsCluster,
-		nv.GetExportPath(),
+	// TODO: use new go-ceph API, see ceph/ceph-csi#2977
+	// new versions of Ceph use a different command, and the go-ceph API
+	// also seems to be different :-/
+	//
+	// run the new command, but fall back to the previous one in case of an
+	// error
+	cmds := [][]string{
+		// ceph nfs export rm <cluster_id> <pseudo_path>
+		{
+			"--id", nv.cr.ID,
+			"--keyfile=" + nv.cr.KeyFile,
+			"-m", nv.mons,
+			"nfs",
+			"export",
+			"rm",
+			nfsCluster,
+			nv.GetExportPath(),
+		},
+		// ceph nfs export delete <cluster_id> <pseudo_path>
+		{
+			"--id", nv.cr.ID,
+			"--keyfile=" + nv.cr.KeyFile,
+			"-m", nv.mons,
+			"nfs",
+			"export",
+			"delete",
+			nfsCluster,
+			nv.GetExportPath(),
+		},
+	}
+
+	var stderr string
+	for _, cmd := range cmds {
+		_, stderr, err = util.ExecCommand(nv.ctx, "ceph", cmd...)
+		if err == nil {
+			break
```

Comment:
instead of running two commands always if the first one fails. check for the `invalid command` error message and do fall back?

Reply:
sure, I can do that